project/src/main/java/com/google/sps/servlets/InvestmentCalculator.java

```diff
@@ -0,0 +1,131 @@
+            while (rs.next()) {
+                amtAvailable = rs.getInt(1);
+            }
+        }
+        return amtAvailable;
+    }
+
+    public int sumInvestmentValues(Connection conn, int userId, int competitionId) throws SQLException{
+        String stmt = "SELECT google_search, invest_date, sell_date, amt_invested FROM investments WHERE user=" + userId 
+        + " AND competition=" + competitionId + ";";
+        int totalInvestmentsValue = 0;
+        try (PreparedStatement investmentsStmt = conn.prepareStatement(stmt);) {
+            // Execute the statement
+            ResultSet rs = investmentsStmt.executeQuery();
+            String googleSearch;
+            long investDate;
+            Date sellDateOrNull;
+            long sellDate;
+            int amtInvested;
+            while (rs.next()) {
+                googleSearch = rs.getString(1);
+                investDate = rs.getDate(2).getTime();
+                sellDateOrNull = rs.getDate(3);
+                if (sellDateOrNull == null) {
+                    sellDate = 0;
+                } else {
+                    sellDate = sellDateOrNull.getTime();
+                }
+                amtInvested = rs.getInt(4);
+                totalInvestmentsValue += getInvestmentValue(googleSearch, investDate, sellDate, amtInvested);
+            }
+        }
+        return totalInvestmentsValue;
+    }
+
+    public int getInvestmentValue(String googleSearch, long investDate, long sellDate, int amtInvested) {
+        if (sellDate != 0) {
+            // Investment has been sold
+            return 0;
+        }
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+        Query<Entity> query = Query.newEntityQueryBuilder()
+            .setKind("TrendsData")
+            .setFilter(PropertyFilter.eq("search_term", googleSearch))
+            .build();
+        QueryResults<Entity> trends = datastore.run(query);
+
+        //store dates as long to prevent y2k in 2038 but immediately convert to string for datastore reasons
+        String startDate = (investDate / 1000L) + "";
+        String currentDate = getLatestDateAsString();
+
+        Entity trend;
+        float startValue;
+        float endValue;
+        float investmentValue = 0;
+
+        while (trends.hasNext()) {
+            trend = trends.next();
+            startValue = (float) trend.getLong(startDate);
+            endValue = (float) trend.getLong(currentDate);
+            investmentValue = amtInvested * (endValue / startValue);
+        }
+        return (int) investmentValue;
+    }
+
+    private String getLatestDateAsString() {
+        //get yesterday's epoch time UTC
+        Calendar c = Calendar.getInstance(TimeZone.getTimeZone("GMT"));
+        c.set(Calendar.HOUR_OF_DAY, 0);
+        c.set(Calendar.MINUTE, 0);
+        c.set(Calendar.SECOND, 0);
+        c.set(Calendar.MILLISECOND, 0);
+        long unixTime = c.getTimeInMillis() / 1000 - (24 * 60 * 60);
```

Comment:
imo it's always nice to change eg `(24 * 60 * 60)` to a variable named something like `ONE_DAY_SECONDS`

Reply:
yep, sorted